src/core_atmosphere/dynamics/mpas_atm_time_integration.F

```diff
@@ -7258,19 +7259,45 @@ subroutine summarize_timestep(domain)
        integer, dimension(:), pointer :: indexToCellID
        integer :: indexMax, indexMax_global
        integer :: kMax, kMax_global
+       integer :: offset_1d ! Offset into a multi-dimensional array, as if it were a contiguous 1-d array
        real (kind=RKIND) :: latMax, latMax_global
        real (kind=RKIND) :: lonMax, lonMax_global
        real (kind=RKIND), dimension(5) :: localVals, globalVals
 
-       real (kind=RKIND) :: spd
+       real (kind=RKIND), dimension(:,:), allocatable :: spd
+       !$acc declare create(spd)
        real (kind=RKIND), dimension(:,:), pointer :: w
        real (kind=RKIND), dimension(:,:), pointer :: u, v, uReconstructZonal, uReconstructMeridional, uReconstructX, uReconstructY, uReconstructZ
        real (kind=RKIND), dimension(:,:,:), pointer :: scalars, scalars_1, scalars_2
 
+       logical :: found_NaN
+
        call mpas_pool_get_config(block % configs, 'config_print_global_minmax_vel', config_print_global_minmax_vel)
        call mpas_pool_get_config(block % configs, 'config_print_detailed_minmax_vel', config_print_detailed_minmax_vel)
        call mpas_pool_get_config(block % configs, 'config_print_global_minmax_sca', config_print_global_minmax_sca)
 
+      ! Fetch pointers early for temporary OpenACC data transfers
+      call mpas_pool_get_subpool(block % structs, 'state', state)
+      call mpas_pool_get_subpool(block % structs, 'diag', diag)
+      call mpas_pool_get_array(state, 'w', w, 2)
+      call mpas_pool_get_array(state, 'u', u, 2)
+      call mpas_pool_get_array(diag, 'v', v)
+      call mpas_pool_get_array(state, 'scalars', scalars, 2)
+      if (config_print_detailed_minmax_vel) then
+         MPAS_ACC_TIMER_START('summarize_timestep [ACC_data_xfer]')
```

Comment:
It would lead to less code (and, in my opinion, more readable code) if we moved the `MPAS_ACC_TIMER_{START,STOP}` macros outside of the conditional blocks, e.g.,
```fortran
     MPAS_ACC_TIMER_START('summarize_timestep [ACC_data_xfer]')
     if (config_print_detailed_minmax_vel) then
        !$acc enter data copyin(w,u,v)
     else if (config_print_global_minmax_vel) then
        !$acc enter data copyin(w,u)
     end if
     if (config_print_global_minmax_sca) then
        !$acc enter data copyin(scalars)
     end if
     MPAS_ACC_TIMER_STOP('summarize_timestep [ACC_data_xfer]')
```